### PR TITLE
[build] Add prettier

### LINKED ISSUE #64: Add prettier

### PR DESCRIPTION
**Pull request by AlexVCS**
this should close issue #64

### CODE REVIEW

**Review by spacey-sooty — CHANGES_REQUESTED** [review]
Add a CI check to verify code is formatted and a script for running formatting

**Comment by AlexVCS** [comment]
> Add a CI check to verify code is formatted and a script for running formatting

for the CI check are you talking about something like this: https://prettier.io/docs/en/precommit ?

**Comment by spacey-sooty** [comment]
> > Add a CI check to verify code is formatted and a script for running formatting
> 
> for the CI check are you talking about something like this: [prettier.io/docs/en/precommit](https://prettier.io/docs/en/precommit) ?

No, a GitHub actions check similar to what was done here https://github.com/CurtinFRC/UnnofficialDriverstation/blob/master/.github/workflows/ci.yml#L10-L38

File: package.json [hunk 1]
```diff
@@ -7,11 +7,16 @@
     "start": "astro dev",
     "build": "astro build",
     "preview": "astro preview",
-    "astro": "astro"
+    "astro": "astro",
+    "format": "pnpm exec prettier . --write"
```
Comment (spacey-sooty) [inline package.json]: Needs an ignore path passed to it; .gitignore should be fine

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -20,6 +20,8 @@ jobs:
       run: pnpm install
     - name: Build
       run: pnpm build
+    - name: Format
```
Comment (spacey-sooty) [inline .github/workflows/ci.yml]: Should be a seperate workflow

**Review by spacey-sooty — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (AlexVCS) [reply package.json]: are you saying that inside the quotes for format I need to add `.gitignore`?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (AlexVCS) [reply .github/workflows/ci.yml]: Can you please explain more? I've not written a github action before. Can it not be under Ubuntu, Windows, or Mac?

Comment (AlexVCS) [reply .github/workflows/ci.yml]: Do you mean a separate yml file?

Comment (spacey-sooty) [reply .github/workflows/ci.yml]: > Can you please explain more? I've not written a github action before. Can it not be under Ubuntu, Windows, or Mac?

No it should be under a seperate one labelled, Format

File: package.json (hunk 1 above)
Comment (spacey-sooty) [reply package.json]: No you need to use the `--ignore-path` option

**Review by spacey-sooty — CHANGES_REQUESTED** [review]
Needs a prettierrc.json as well

**Comment by spacey-sooty** [comment]
Stale with unaddressed comments

**Comment by AlexVCS** [comment]
> Stale with unaddressed comments

Does anything need to be changed? I thought I didn't need the `ignore-path` option. It seems like this is in line with that: https://prettier.io/docs/en/cli#--ignore-path

**Comment by spacey-sooty** [comment]
> > Stale with unaddressed comments
> 
> Does anything need to be changed? I thought I didn't need the `ignore-path` option. It seems like this is in line with that: [prettier.io/docs/en/cli#--ignore-path](https://prettier.io/docs/en/cli#--ignore-path)

I'd prefer it being explicitly stated, it makes it more obvious.

**Review by spacey-sooty — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -48,5 +48,19 @@ jobs:
...
+    
+    steps:
+    - uses: actions/checkout@v3
+    - uses: pnpm/action-setup@v3
+      with:
+        version: latest        
+    - name: Install
+      run: pnpm install
+    - name: Run Prettier
+        run: pnpm format
```
Comment (spacey-sooty) [inline .github/workflows/ci.yml]: this indentation is wrong

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -34,7 +34,7 @@ jobs:
       run: pnpm install
     - name: Build
       run: pnpm build
-
+
```
Comment (spacey-sooty) [inline .github/workflows/ci.yml]:
```suggestion

```

**Review by spacey-sooty — CHANGES_REQUESTED** [review]

**Review by spacey-sooty — CHANGES_REQUESTED** [review]
Run the formatter

**Review by spacey-sooty — APPROVED** [review]

**Pull request merged**